### PR TITLE
[FLASK] Add snaps insight UI

### LINKED ISSUE #655: Integrate Transaction Insight into the MetaMask Extension UI
Following https://github.com/MetaMask/snaps-skunkworks/pull/642 we should integrate the transaction insight API into the extension UI.

### PR DESCRIPTION
**Pull request by GuillaumeRx**
## Explanation

Add snaps insight UI tab in contract interactions confirm screen.

## More Information

* Fixes [#655](https://github.com/MetaMask/snaps-skunkworks/issues/655)

### CODE REVIEW

**Review by hmalik88 — APPROVED** [review]
LGTM.

File: ui/hooks/flask/useTransactionInsightSnap.js [hunk 1]
```diff
@@ -5,28 +5,33 @@ import { getPermissionSubjects } from '../../selectors';
...
   useEffect(() => {
     async function fetchInsight() {
       const d = await handleSnapRequest({
-        snapId, origin: undefined, handler: 'onTransaction', request: {
-          params: [transaction],
-        }
+        snapId,
+        origin: 'test',
+        handler: 'onTransaction',
+        request: {
```
Comment (hmalik88) [inline ui/hooks/flask/useTransactionInsightSnap.js:23]: Is the request object validated to have a method property? If not, we can remove this.

Comment (GuillaumeRx) [reply ui/hooks/flask/useTransactionInsightSnap.js:23]: Yes, it's expecting a proper JSON-RPC method so it's the only way to make it work

File: ui/components/ui/tabs/dropdown-tab/dropdown-tab.component.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
```
Comment (brad-decker) [inline ui/components/ui/tabs/dropdown-tab/dropdown-tab.component.js:1]: Please remove the .component extension in the filename, this is an old pattern from a time  when we had .containers too but now that we use hooks we do not create container files.

File: ui/components/app/confirm-page-container/flask/snap-insight.component.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import React from 'react';
```
Comment (brad-decker) [inline ui/components/app/confirm-page-container/flask/snap-insight.component.js:1]: Remove the .component extension on the filename please

**Review by brad-decker — CHANGES_REQUESTED** [review]
filename nits mostly, looks good otherwise

**Pull request merged**